openqabot/approver.py

```diff
@@ -131,6 +132,53 @@ def is_job_marked_acceptable_for_incident(self, job_id: int, inc: int) -> bool:
             pass
         return False
 
+    @lru_cache(maxsize=512)
+    def was_green_before(self, job_id: int, inc: int) -> bool:
+        log.info("Current aggregate test %s failed. Will look at older jobs" % job_id)
+        # We need a considerable amount of older jobs, since there could be many failed manual restarts from same day
+        older_jobs = self.client.get_older_jobs(job_id, 20)
+
+        current_build = older_jobs['data'][0]['build'].replace('-1','')
+        log.info(current_build)
+        current_build_date = date(
+            int(current_build[0:4]),
+            int(current_build[4:6]),
+            int(current_build[6:8])
+            )
+        log.info("Failed job date: %s",current_build_date)
+
+        # Use at most X days old build. Don't go back in time too much to reduce risk of using invalid tests
+        oldest_build_usable = current_build_date - timedelta(days=6)
```

Comment:
ok, that makes sense in general but why 6 then?

I suggest to put that magic number into a configuration variable or a constant with a good name and the according comment for the value selected.

Reply:
Same as the decision of asking for 20 older jobs to openQA and not 15 or 25,  I don't see a reasonable number but a "bunch of tests" to be collected.

For the days: more days means more chance to find all green test at some point (but useless if update request is new). I considered reasonable a week, even though more might be useful (some updates might be stuck for quite some time).

 But more days means also more risk of taking a "unfaithful" test, where in meantime the update request was changed this opens the pandora's box since it's an issue that is already present, and incident would get retriggered anyway so they provide some safety. 
 
 Anyway, I agree on making this number more visible, can move it [here maybe](https://github.com/openSUSE/qem-bot/blob/master/openqabot/__init__.py)